src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreaker.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.circuitbreaker;
+
+/**
+ * A <a href="http://martinfowler.com/bliki/CircuitBreaker.html">circuit breaker</a>, which tracks the number of
+ * succcess/failure requests and detects a remote service failure.
+ */
+public interface CircuitBreaker {
```

Comment:
Could we add a static factory method to this interface so that creating a circuit breaker with the default settings is simple?

```java
CircuitBreaker.of("name");
CircuitBreaker.ofDefaultName();
```